src/App.tsx

```diff
@@ -1,14 +1,62 @@
-/* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
+import { Todo } from './types/Todo';
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
+import { getTodos } from './api';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Select } from './types/Select';
+
+const getFilteredTodos = (
+  todos: Todo[],
+  query: string,
+  selectedFilter: string,
+) => {
+  let todosCopy = [...todos];
+
+  if (query) {
+    todosCopy = todos.filter(({ title }) => {
+      return title.toLowerCase().includes(query.toLowerCase());
+    });
+  }
+
+  if (selectedFilter) {
+    switch (selectedFilter) {
+      case 'active':
+        return todosCopy.filter(({ completed }) => !completed);
+      case 'completed':
+        return todosCopy.filter(({ completed }) => completed);
+      default:
+        return todosCopy;
+    }
+  }
+
+  return todosCopy;
+};
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [query, setQuery] = useState('');
+  const [selectedFilter, setSelectedFilter] = useState<Select>(Select.all);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    getTodos().then((todosFrommServer) => {
+      setTodos(todosFrommServer);
+    }).catch((error) => {
+      // eslint-disable-next-line no-console
+      console.log(error);
+    }).finally(() => setIsLoading(false));
```

Comment:
```suggestion
    getTodos()
      .then((todosFrommServer) => {
        setTodos(todosFrommServer);
      })
      .catch((error) => {
        // eslint-disable-next-line no-console
        console.log(error);
      })
      .finally(() => setIsLoading(false));
```
Don't forget about code style